src/chain_rules.jl

```diff
@@ -18,10 +18,10 @@ end
 function ChainRulesCore.frule((_, Δp, _), ::typeof(differentiate), p, x)
     return differentiate(p, x), differentiate(Δp, x)
 end
-function pullback(Δdpdx, x)
+function pullback_differentiate_polynomial(Δdpdx, x)
     return ChainRulesCore.NoTangent(), x * differentiate(x * Δdpdx, x), ChainRulesCore.NoTangent()
 end
-function ChainRulesCore.rrule(::typeof(differentiate), p, x)
+function ChainRulesCore.rrule(::typeof(differentiate), p::APL, x)
```

Comment:
Is that resolving an issue you had ? `differentiate` is defined by this package so I don't think it would give issue to not specify the type here